src/storage/hummock_sdk/src/table_watermark.rs

```diff
@@ -593,6 +564,116 @@ impl TableWatermarks {
     }
 }
 
+impl ProtoSerializeSizeEstimatedExt for TableWatermarks {
+    fn estimated_encode_len(&self) -> usize {
+        self.watermarks.len() * size_of::<HummockEpoch>()
+            + self
+                .watermarks
+                .iter()
+                .map(|(_, watermarks)| {
+                    watermarks
+                        .iter()
+                        .map(|watermark| watermark.estimated_size())
+                        .sum::<usize>()
+                })
+                .sum::<usize>()
+            + size_of::<bool>() // for direction
+    }
+}
+
+impl ProtoSerializeExt for TableWatermarks {
+    type PB = PbTableWatermarks;
+    type T = TableWatermarks;
+
```

Comment:
If the two implementations are the same, we can put them in a common method and call the method in the trait.